.buildkite/steps/check-code-committed.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env sh
+
+set -euf
+
+echo --- :go: Checking go mod tidyness
+go mod tidy
+if ! git diff --no-ext-diff --exit-code; then
+  echo ^^^ +++
+  echo "The go.mod or go.sum files are out of sync with the source code"
+  echo "Please run \`go mod tidy\` locally, and commit the result."
+
+  exit 1
+fi
+
+echo --- :go: Checking go formatting
+gofmt -w .
```

Comment:
optional, but it could be nice/more flexible to install golangci-lint, which also comes with a `--diff` flag that will exit non-zero with a report of issues